### ISSUE TITLE
Noticeable quality degradation in low light situations.

### ISSUE THREAD

**Issue by solonovamax** [aka-katto/dandere2x#151]
Too lazy to write up a full issue with all the fancy stuff with headers and whatever, so here's some basic stuff:

Here is a screenshot of the frame before (note: is screenshot and I did _not_ extract the raw frame from the video using ffmpeg.)
![image](https://user-images.githubusercontent.com/46940694/96660158-d6453480-1316-11eb-8d20-8bcfc4f7ce54.png)

Frame from after upscaling (note: this is on a 1080p monitor.)
![image](https://user-images.githubusercontent.com/46940694/96660207-fb39a780-1316-11eb-8d00-f7ccaa5a281c.png)

(And yes, I recognize the quality of the first image wasn't insane. The second one is just significantly worse with all that noise.)

Also, similar to #149, the grid is super noticeable here.
It seems this is visible mostly in low light scenes.

These frames were taken from the short film Shelter, at roughly 0:00:50

This is probably from the fact that it's being JPEG encoded and decoded, but this could easily be fixed if you were to just save the raw files to disk, rather than JPEG encoding them. (Who cares if it takes up even a whole gigabyte. Storage is cheap these days and it'll be deleted after anyways.)
Hell, just write the raw byte array of pixel values to disk without any file ending, then read them as needed.

**Comment by aka-katto** [comment]
I haven't tested png in a while, but you can change it here:

https://github.com/aka-katto/dandere2x/blob/master/src/dandere2x_win32.yaml

on line 50 in the yaml.

From my testing (two years ago now, wow), png / jpg didn't provide much of a difference (psnr testing) - if memory serves me right. I'll run some tests once I'm finished moving on the PSNR difference between them again

**Comment by solonovamax** [comment]
I'll run a few tests with the first few seconds of that clip (whole thing takes too long) when I get a chance, but it might be the jpeg compression or it might be whatever settings you guys have for ffmpeg.

**Comment by solonovamax** [comment]
After fucking around with a lot of random shit, it seems the file format doesn't matter. After changing it to png (and making sure it outputs png files for everything), it seems to still look like shit.
(Also, the setup procedure is lacking. Took me a while to figure out it only works if the root dir is the src dir, meaning I can't just clone it directly. That should be in a title font at the top of the page.)

**Comment by solonovamax** [comment]
Update, I've traced it down to the [re_encode_video](https://github.com/aka-katto/dandere2x/blob/master/src/wrappers/ffmpeg/ffmpeg.py#L42) method.

**Comment by solonovamax** [comment]
UpdateV2: [why are you adding noise to the video](https://github.com/aka-katto/dandere2x/blob/master/src/dandere2x_linux.yaml#L164)...

**Comment by solonovamax** [comment]
so the noise decreases how long it takes to upscale it?????

**Comment by solonovamax** [comment]
idk what this dark magic is, but it seems switching to the luminosity channel (channel 0) makes it less noticeable, but keeps the performance increase?
I'll make a pull request now, but I'll do more testing to see if this is actually the case later.

**Comment by aka-katto** [comment]
Left a comment in the MR - closing it for now since it's "closed", just pending

**Comment by aka-katto** [comment]
> 
> 
> UpdateV2: [why are you adding noise to the video](https://github.com/aka-katto/dandere2x/blob/master/src/dandere2x_linux.yaml#L164)...


Noise is added to the video to normalize the data - without this, our statistical functions used for block matching would have issues performing matches. The noise itself is removed by the upscaler (can confirm that it seems to be indistinguishable). I've had ideas / thoughts about moving over this noising-of-data to the cpp element, but free time is nnn

tl;dr you dont have to worry about the noise causing issues for the actual output, so-long-as the waifu2x denoise is above 1
